sklearn/linear_model/_glm/glm.py

```diff
@@ -0,0 +1,678 @@
+
+            For ``0<power<1``, no distribution exists.
+
+            Special cases are:
+
+            +-------+------------------+
+            | Power | Distribution     |
+            +=======+==================+
+            | 0     | Normal           |
+            +-------+------------------+
+            | 1     | Poisson          |
+            +-------+------------------+
+            | (1,2) | Compound Poisson |
+            +-------+------------------+
+            | 2     | Gamma            |
+            +-------+------------------+
+            | 3     | Inverse Gaussian |
+            +-------+------------------+
+
+    alpha : float, optional (default=1)
+        Constant that multiplies the penalty terms and thus determines the
+        regularization strength.
+        See the notes for the exact mathematical meaning of this
+        parameter.``alpha = 0`` is equivalent to unpenalized GLMs. In this
+        case, the design matrix X must have full column rank
+        (no collinearities).
+
+    link : {'auto', 'identity', 'log'}, default='auto'
+        The link function of the GLM, i.e. mapping from linear predictor
+        (X*coef) to expectation (y_pred). Option 'auto' sets the link
+        depending on the chosen family as follows:
+
+        - 'identity' for Normal distribution
+
+        - 'log' for Poisson,  Gamma or Inverse Gaussian distributions
+
+    fit_intercept : boolean, optional (default=True)
+        Specifies if a constant (a.k.a. bias or intercept) should be
+        added to the linear predictor (X*coef+intercept).
+
+    max_iter : int, optional (default=100)
+        The maximal number of iterations for the solver.
+
+    tol : float, optional (default=1e-4)
+        Stopping criterion. For the lbfgs solver,
+        the iteration will stop when ``max{|g_i|, i = 1, ..., n} <= tol``
+        where ``g_i`` is the i-th component of the gradient (derivative) of
+        the objective function.
+
+    warm_start : boolean, optional (default=False)
+        If set to ``True``, reuse the solution of the previous call to ``fit``
+        as initialization for ``coef_`` and ``intercept_`` .
+
+    copy_X : boolean, optional, (default=True)
+        If ``True``, X will be copied; else, it may be overwritten.
+
+    verbose : int, optional (default=0)
+        For the lbfgs solver set verbose to any positive number for verbosity.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,) Estimated coefficients for the linear
+        predictor (X*coef_+intercept_) in the GLM.
+
+    intercept_ : float
+        Intercept (a.k.a. bias) added to linear predictor.
+
+    n_iter_ : int
+        Actual number of iterations used in the solver.
+    """
+    def __init__(self, power=0.0, alpha=1.0, fit_intercept=True,
+                 link='auto', max_iter=100, tol=1e-4,
+                 warm_start=False, copy_X=True, check_input=True, verbose=0):
```

Comment:
we are not consistent here with check_input. Elsewhere it's used as a fit param. I would suggest to remove this parameter from the public API.